File: app/src/main/java/com/sumian/app/network/response/ErrorResponse.java
package com.sumian.app.network.response;

/**
 * Created by jzz
 * on 2017/9/30
 * <p>
 * desc:
 */

public class ErrorResponse {

    private int status_code;
    private String message;

    public int getStatus_code() {
        return status_code;
    }

    public void setStatus_code(int status_code) {
        this.status_code = status_code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Override
    public String toString() {
        return "ErrorResponse{" +
            "status_code=" + status_code +
            ", message='" + message + '\'' +
            '}';
    }
}

File: common/src/main/java/com/sumian/common/network/response/ErrorResponse.kt
package com.sumian.common.network.response

import com.sumian.common.network.error.ErrorInfo400
import com.sumian.common.network.error.ErrorInfo499

/**
 * <pre>
 *     @author : Zhan Xuzhao
 *     e-mail : xuzhao.z@sumian.com
 *     time   : 2018/6/14 10:29
 *     desc   :
 *     version: 1.0
 * </pre>
 */
data class ErrorResponse(var code: Int, var message: String) {

    companion object {
        @JvmStatic
        fun createFromErrorInfo(errorInfo: ErrorInfo499?): ErrorResponse? {
            if (errorInfo == null) {
                return null
            }
            return ErrorResponse(errorInfo.error.code, errorInfo.error.userMessage)
        }

        @JvmStatic
        fun createFromErrorInfo(errorInfo: ErrorInfo400?): ErrorResponse? {
            if (errorInfo == null) {
                return null
            }
            return ErrorResponse(errorInfo.statusCode, errorInfo.message)
        }
    }
}





